frontend/packages/kubevirt-plugin/src/selectors/pod/selectors.ts

```diff
@@ -42,15 +42,22 @@ export const findPodWithOneOfStatuses = (pods: PodKind[], statuses: string[]) =>
     return statuses.some((status) => status === phase);
   });
 
-export const findVMPod = (pods: PodKind[], vm: VMKind, podNamePrefix: string) => {
+export const findVMPod = (pods: PodKind[], vm: VMKind) => {
   if (!pods) {
     return null;
   }
-  const prefix = `${podNamePrefix}${getName(vm)}-`;
-  const prefixedPods = pods.filter(
-    (p) => getNamespace(p) === getNamespace(vm) && getName(p).startsWith(prefix),
+
+  let prefixedPods = pods.filter(
+    (p) => getNamespace(p) === getNamespace(vm) && getOwnerReferences(p)[0].name === getName(vm),
   );
 
+  if (prefixedPods.length > 1) {
+    prefixedPods = [
+      prefixedPods.sort((a: PodKind, b: PodKind) =>
+        a.metadata.creationTimestamp > b.metadata.creationTimestamp ? 1 : -1,
+      )[0],
+    ];
+  }
   return (
```

Comment:
we don't have to check for the statuses anymore, just return the pod with the newest timestamp